src/Estimators/tests/RobotDynamicsEstimator/UkfStateTest.cpp

```diff
@@ -0,0 +1,192 @@
+    double dT;
+    REQUIRE(parameterHandler->getGroup("GENERAL").lock()->getParameter("sampling_time", dT));
+    groupUKFState->setParameter("sampling_time", dT);
+
+    std::unique_ptr<UkfState> stateModel = UkfState::build(groupUKFState,
+                                                           kinDyn,
+                                                           subModelList,
+                                                           kinDynWrapperList);
+
+    // Create an input for the ukf state
+    UKFInput input;
+
+    // Define joint positions
+    Eigen::VectorXd jointPos;
+    jointPos.resize(kinDyn->model().getNrOfDOFs());
+    jointPos.setZero();
+    input.robotJointPositions = jointPos;
+
+    // Define base pose
+    manif::SE3d basePose;
+    basePose.setIdentity();
+    input.robotBasePose = basePose;
+
+    // Define base velocity and acceleration
+    manif::SE3d::Tangent baseVelocity, baseAcceleration;
+    baseVelocity.setZero();
+    baseAcceleration.setZero();
+    input.robotBaseVelocity = baseVelocity;
+    input.robotBaseAcceleration = baseAcceleration;
+
+    std::shared_ptr<UkfInputProvider> inputProvider = std::make_shared<UkfInputProvider>();
+
+    BipedalLocomotion::System::VariablesHandler stateHandler = stateModel->getStateVariableHandler();
+    int stateSize = stateModel->getStateSize();
+
+    stateModel->setUkfInputProvider(inputProvider);
+
+    Eigen::VectorXd currentState;
+    currentState.resize(stateSize);
+    currentState.setZero();
+
+    Eigen::VectorXd motorTorques;
+    motorTorques.resize(kinDyn->model().getNrOfDOFs());
+    motorTorques << -1.6298, -1.10202, 0, -0.74, 0.0877, -0.00173;
+    Eigen::VectorXd wrenchFTtLeg;
+    wrenchFTtLeg.resize(6);
+    wrenchFTtLeg << 0, 0, 100.518, 0.748, 0.91, 0;
+    Eigen::VectorXd wrenchFTFootFront;
+    wrenchFTFootFront.resize(6);
+    wrenchFTFootFront << 0, 0, 1.761, -0.001, 0.0003, 0;
+    Eigen::VectorXd wrenchFTFootRear;
+    wrenchFTFootRear.resize(6);
+    wrenchFTFootRear << 0, 0, 1.752, 0.000876, 0.000649, 0;
+
+    currentState.segment(stateHandler.getVariable("tau_m").offset, stateHandler.getVariable("tau_m").size) = motorTorques;
+    currentState.segment(stateHandler.getVariable("r_leg_ft_sensor").offset, stateHandler.getVariable("r_leg_ft_sensor").size) = wrenchFTtLeg;
+    currentState.segment(stateHandler.getVariable("r_foot_front_ft_sensor").offset, stateHandler.getVariable("r_foot_front_ft_sensor").size) = wrenchFTFootFront;
+    currentState.segment(stateHandler.getVariable("r_foot_rear_ft_sensor").offset, stateHandler.getVariable("r_foot_rear_ft_sensor").size) = wrenchFTFootRear;
+
+    Eigen::VectorXd updatedState;
+    updatedState.resize(stateSize);
+
+    REQUIRE(inputProvider->setInput(input));
+
+    stateModel->propagate(currentState, updatedState);
+
+    for (int idx = 0; idx < updatedState.size(); idx++)
+    {
+        REQUIRE(std::abs(updatedState[idx]) < 200);
+    }
+}
```

Comment:
I would also suggest defining a test case where you expect some specific output, and check that you converge there. You could generate in some way some simple synthetic data (i.e. a set of constant joint torques), and check that the estimator converges to the quantities you used to generate the data (i.e. you check that you are able to reconstruct the initial set of joint torques)